### ISSUE TITLE
Many compiler warnings

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#308]
With recent changes to deal.II, we now also compile ASPECT with many more `-W...` flags on. This triggers a large number of warnings.

Patch for most of them forthcoming.

**Comment by bangerth** [comment]
The ones I don't know what to do with are of the form
```
source/simulator/helper_functions.cc:855:153: warning: 'dealii::types::boundary_id dealii::TriaAccessor<<anonymous>, <anonymous>, <anonymous> >::boundary_indicator() const [with int structdim = 2; int dim = 3; int spacedim = 3; dealii::types::boundary_id = unsigned char]' is deprecated (declared at /home/bangerth/p/deal.II/1/install/include/deal.II/grid/tria_accessor.templates.h:1841) [-Wdeprecated-declarations]
```
The deprecation (and the provision of an alternative function and type) postdate the last deal.II release. In other words, I cannot fix the warning without breaking backward compatibility with the last deal.II release.

@tjhei : What do you want to do here? I know you'd like to keep backward compatibility for now.

**Comment by tjhei** [comment]
I would like to be compatible with the latest release if possible. We can:
a) ignore the warnings (this is for people running the development version of deal.II anyways)
b) add a bunch of #ifdefs
c) disable warnings when compiling ASPECT.

My vote is for b) and aggressively clean them up when a new deal.II release is made. If you agree, I can tackle them.

**Comment by bangerth** [comment]
OK, if you want b) then I will gladly let you do it :-) The one that shows up in large numbers is
```
  cell->boundary_indicator()
```
which should now be
```
  cell->boundary_id()
```
I think #315 takes care of most of the other warnings.

**Comment by bangerth** [comment]
There are 14 such cases, btw:
```
> grep -r boundary_indicator include/ source/ | grep cell
source/simulator/helper_functions.cc:              (parameters.prescribed_velocity_boundary_indicators.find(cell->face(f)->boundary_indicator())!=
source/simulator/helper_functions.cc:                    velocity_boundary_conditions.find(cell->face(f)->boundary_indicator())->second->boundary_velocity(fe_face_values.quadrature_point(q));
source/simulator/freesurface.cc:               ((sim.parameters.free_surface_boundary_indicators.find(cell->face(face_no)->boundary_indicator())
source/simulator/freesurface.cc:             sim.parameters.free_surface_boundary_indicators.find(cell->face(face_no)->boundary_indicator())!=
source/geometry_model/africa_region.cc:            cell->face(f)->set_all_boundary_indicators(f);
source/geometry_model/africa_region.cc:                cell->face(f)->boundary_indicator() == 5) // face is at top boundary
source/postprocess/mass_flux_statistics.cc:                local_boundary_fluxes[cell->face(f)->boundary_indicator()]
source/postprocess/heat_flux_statistics.cc:                local_boundary_fluxes[cell->face(f)->boundary_indicator()]
source/postprocess/topography.cc:            if (cell->face(face_no)->at_boundary() && cell->face(face_no)->boundary_indicator() == relevant_boundary )
source/postprocess/velocity_boundary_statistics.cc:                local_max_vel[cell->face(f)->boundary_indicator()] = std::max(local_max,
source/postprocess/velocity_boundary_statistics.cc:                                                                              local_max_vel[cell->face(f)->boundary_indicator()]);
source/postprocess/velocity_boundary_statistics.cc:                local_min_vel[cell->face(f)->boundary_indicator()] = std::min(local_min,
source/postprocess/velocity_boundary_statistics.cc:                                                                              local_min_vel[cell->face(f)->boundary_indicator()]);
source/postprocess/table_heat_flux_statistics.cc:                local_boundary_fluxes[cell->face(f)->boundary_indicator()]
```

**Comment by gassmoeller** [comment]
Is this closed by #315 and #322? I get no more warnings with a recent dev version of deal.II.

**Comment by tjhei** [comment]
Yes, thanks.
